edx_django_utils/cache/utils.py

```diff
@@ -335,25 +338,18 @@ def __nonzero__(self):
     def __bool__(self):
         raise CachedResponseError()
 
-    def __index__(self):
-        raise CachedResponseError()
-
-    def __getattr__(self, name):
-        raise CachedResponseError()
-
-    def __setattr__(self, name, val):
-        if name not in self.VALID_ATTRIBUTES:
-            raise CachedResponseError()
-        return super(CachedResponse, self).__setattr__(name, val)
-
-    def __getitem__(self, key):
-        raise CachedResponseError()
-
-    def __setitem__(self, key, val):
+    def __eq__(self, other):
+        if isinstance(other, CachedResponse):
+            if self.is_hit != other.is_hit:
+                return False
+            if self.key != other.key:
+                return False
+            if self.is_hit:
+                if self.value != other.value:
+                    return False
+            return True
```

Comment:
I realize why you did it this way, and I think it's correct. But it seems pretty awkward to follow. Maybe it'd be clearer with something like:

```python
def __eq__(self, other):
    if not isinstance(other, CachedResponse):
        raise CachedResponseError()
    if self.is_hit != other.is_hit:
        return False

    if self.is_hit:
        return (self.key, self.value) == (other.key, other.value)
    else:
        return self.key == other.key  # cache misses have no value attribute
```

?

A bit of a nit, but it does seem a little weird to have an `__init__` that yields a object with different attributes set depending on values. I feel like that works great for languages that actually support algebraic types, but it causes these kinds of weird contortions in Python. If cache misses still had `None` for a value, this whole thing could have been:

```python
def __eq__(self, other):
    if not isinstance(other, CachedResponse):
        raise CachedResponseError()
    return (self.is_hit, self.key, self.value) == (self.is_hit, other.key, other.value)
```

Which I think is the most natural way to do it, and doesn't seem that risky in terms of foot-gun behavior.